scalableminds/webknossos
# Loading spinner for some long requests

This is in update to the last PR (#1615) for this. 😊  I noticed that it wasn't working as expected on the master. So I made two changes:

1. Use the correct life cycle event for "exp. tracing". 
2. The statistics tab didn't show the spinner because it was hidden by some else:
   1. Router handles route for Weekly Statistics View
   2. Router initializes Statistics View
   3. Statistics View shows spinner and starts the request to fetch the model
   4. Router call `changeView` which hides the spinner :rage4: 
   5. Long Request returns, but spinner is already hidden ...

I believe we can eliminate `hideLoadingSpinner`within `changeView` since all other router functions usually call `hideLoadingSpinner` themselves.

Steps to test:
- Go to "Weekly Statistics View" and see spinner while loading
- Go to "Explorative Tracing Tab" and see spinner while loading
- (throttle connection, or go offline when switching to either view)

Issues:
- fixes #1355

------


<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1648/create?referer=github" target="_blank">Log Time</a>
